src/gt4py/next/otf/binding/cpp_interface.py

```diff
@@ -29,63 +26,33 @@
     header_extension="cpp.inc",
 )
 
-_TYPE_MAPPING: Final = types.MappingProxyType(
-    {
-        bool: "bool",
-        int: "long",
-        float: "double",
-        complex: "std::complex<double>",
-        np.bool_: "bool",
-        np.byte: "signed char",
-        np.ubyte: "unsigned char",
-        np.short: "short",
-        np.ushort: "unsigned short",
-        np.intc: "int",
-        np.uintc: "unsigned int",
-        np.int_: "long",
-        np.uint: "unsigned long",
-        np.longlong: "long long",
-        np.ulonglong: "unsigned long long",
-        np.single: "float",
-        np.double: "double",
-        np.longdouble: "long double",
-        np.csingle: "std::complex<float>",
-        np.cdouble: "std::complex<double>",
-        np.clongdouble: "std::complex<long double>",
-        ctypes.c_bool: "bool",
-        ctypes.c_char: "char",
-        ctypes.c_wchar: "wchar_t",
-        ctypes.c_byte: "char",
-        ctypes.c_ubyte: "unsigned char",
-        ctypes.c_short: "short",
-        ctypes.c_ushort: "unsigned short",
-        ctypes.c_int: "int",
-        ctypes.c_uint: "unsigned int",
-        ctypes.c_long: "long",
-        ctypes.c_ulong: "unsigned long",
-        ctypes.c_longlong: "long long",
-        ctypes.c_ulonglong: "unsigned long long",
-        ctypes.c_size_t: "std::size_t",
-        ctypes.c_ssize_t: "std::ptrdiff_t",
-        ctypes.c_float: "float",
-        ctypes.c_double: "double",
-        ctypes.c_longdouble: "long double",
-    }
-)
-
 
-def render_python_type(python_type: Type) -> str:
-    return _TYPE_MAPPING[python_type]
+def render_scalar_type(scalar_type: ts.ScalarType) -> str:
+    if scalar_type.kind == ts.ScalarKind.BOOL:
+        return "bool"
+    elif scalar_type.kind == ts.ScalarKind.INT32:
+        return "std::int32_t"
+    elif scalar_type.kind == ts.ScalarKind.INT64:
+        return "std::int64_t"
+    elif scalar_type.kind == ts.ScalarKind.FLOAT32:
+        return "float"
+    elif scalar_type.kind == ts.ScalarKind.FLOAT64:
+        return "double"
+    elif scalar_type.kind == ts.ScalarKind.STRING:
+        return "std::string"
+    elif scalar_type.kind == ts.ScalarKind.DIMENSION:
+        raise AssertionError(f"Deprecated type '{scalar_type}' is not supported.")
+    else:
+        raise AssertionError(f"Scalar kind '{scalar_type}' is not implemented when it should be.")
```

Comment:
Using a dict wasn't such a bad idea (even if the original one had too many cases), but otherwise, I think using pattern matching makes the code more readable here:

```suggestion
    match scalar_type.kind:
        case ts.ScalarKind.BOOL:
            return "bool"
        case ts.ScalarKind.INT32:
            return "std::int32_t"
        case ts.ScalarKind.INT64:
            return "std::int64_t"
        case ts.ScalarKind.FLOAT32:
            return "float"
        case ts.ScalarKind.FLOAT64:
            return "double"
        case ts.ScalarKind.STRING:
            return "std::string"
        case ts.ScalarKind.DIMENSION:
            raise AssertionError(f"Deprecated type '{scalar_type}' is not supported.")
        case _:
            raise AssertionError(f"Scalar kind '{scalar_type}' is not implemented when it should be.")
```